tasks/eva_2121/aggregate_table.py

```diff
@@ -0,0 +1,226 @@
+    reports = glob.glob(os.path.join(accession_dir, '*_assembly_report.txt'))
+    if fastas and len(fastas) == 1:
+        fasta = fastas[0]
+    elif nucleotide_fastas and len(nucleotide_fastas) == 1:
+        fasta = nucleotide_fastas[0]
+    if reports and len(reports) == 1:
+        report = reports[0]
+
+    if fasta is None or report is None:
+        report, fasta = download_assembly(scientific_name, assembly_accession, download_dir)
+    return fasta, report
+
+
+def download_assembly(scientific_name, assembly_accession, download_dir):
+    python = os.path.join(eva_accession_path, "python-import-automation/bin/python")
+    genome_downloader = os.path.join(eva_accession_path, "eva-accession/eva-accession-import-automation/genome_downloader.py")
+    private_json = os.path.join(eva_accession_path, "private-config.json")
+
+    output_dir = os.path.join(download_dir, scientific_name.lower().replace(' ', '_'))
+    os.makedirs(output_dir, exist_ok=True)
+    command = "{} {} -p {} -a {} -o {}""".format(python, genome_downloader, private_json, assembly_accession, output_dir)
+    run_command_with_output('Retrieve assembly fasta and assembly report', command)
+
+    assembly_report = glob.glob(os.path.join(output_dir, assembly_accession, "*_assembly_report.txt"))[0]
+    assembly_fasta = os.path.join(output_dir, assembly_accession, assembly_accession + ".fa")
+    return assembly_fasta, assembly_report
+
+
+def count_variants(rows, only_variant_to_process=True):
+    """Sum variants in the rows provided."""
+    stay_unchanged = {row['Will stay unchanged'] for row in rows}
+    return sum([
+        int(row['Number Of Variants (submitted variants)'].replace(',', ''))
+        for row in rows
+        if not only_variant_to_process or ('no' in stay_unchanged and row['Source'] != 'DBSNP - filesystem')
+    ])
+
+
+def aggregate_list_of_species(input_file, assembly_dir, download_dir, output_assemblies_tsv, output_taxonmomy_tsv):
+    """
+    """
+    data_per_taxid, data_per_taxid_and_assembly = parse_input(input_file)
+    only_unmapped_tax_id = []
+    unchanged_taxid = []
+    dbsnps_only = []
+    eva_only = []
+    mixture = []
+    taxid_to_tempmongo = {}
+    total_number_variant = 0
+    for taxid in data_per_taxid:
+        rows = data_per_taxid[taxid]
+        sources = {row['Source'] for row in rows}
+        stay_unchanged = {row['Will stay unchanged'] for row in rows}
+
+        number_variants = count_variants(rows)
+        total_number_variant += number_variants
+
+        source = None
+        if len(sources) == 1:
+            source = sources.pop()
+        if source == 'DBSNP - filesystem':
+            only_unmapped_tax_id.append(taxid)
+        elif 'no' not in stay_unchanged:
+            unchanged_taxid.append(taxid)
+        elif source == 'EVA':
+            eva_only.append(taxid)
+        elif source == 'DBSNP':
+            dbsnps_only.append(taxid)
+        else:
+            mixture.append(taxid)
```

Comment:
I don't see any such category in the sheet...

Reply:
They are categories per taxonomy to detect which taxonomy needs to be processed and which does not. the `mixture` is where this taxonomy has data coming from at least two of dbsnp, eva and unmapped.